rpc/src/rpc_subscriptions.rs

```diff
@@ -573,15 +573,18 @@ impl RpcSubscriptions {
         let blockstore = Blockstore::open(&ledger_path).unwrap();
         let blockstore = Arc::new(blockstore);
 
-        Self::new_with_config(
+        let rpc_subscriptions = Self::new_with_config(
             exit,
             max_complete_transaction_status_slot,
             blockstore,
             bank_forks,
             block_commitment_cache,
             optimistically_confirmed_bank,
             &PubSubConfig::default_for_tests(),
-        )
+        );
+        // Ensure RPC notifier is ready to receive notifications before proceeding
```

Comment:
Hmm, this doesn't really ensure it's ready to receive notifications, does it? I'd prefer a deterministic check, if possible.